test/IntegrationTests/TicketService/test_create_custom_fields.py

```diff
@@ -0,0 +1,160 @@
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "test_field",
+            "field_type": "VARCHAR(255)",
+            "required": True
+        }
+        logger.info(f"Testing valid custom field addition with data: {data}")
+        response = requests.post(url, json=data)
+        self.assertEqual(response.status_code, 200, "Failed to add custom field")
+        response_data = response.json()
+        self.assertEqual(response_data["field_name"], "test_field")
+        logger.info("Test case for valid custom field addition passed.")
+
+    def test_add_custom_field_missing_required_field(self):
+        """Test adding a custom field with missing required field."""
+        url = f"{self.BASE_URL}/custom_fields"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "missing_required_field",
+            "field_type": "VARCHAR(255)"
+            # 'required' field is missing
+        }
+        logger.info(f"Testing missing 'required' field with data: {data}")
+        response = requests.post(url, json=data)
+        self.assertEqual(response.status_code, 422, "Missing required field should result in 422")
+        self.assertIn("field required", response.text.lower())
+        logger.info("Test case for missing required field passed.")
+
+    def test_add_custom_field_invalid_customer_guid(self):
+        """Test adding a custom field with invalid customer_guid."""
+        url = f"{self.BASE_URL}/custom_fields"
+        data = {
+            "customer_guid": "256128e2-b963-4019-b870-d6e82db0d631", # Invalid or random customer_guid
+            "field_name": "test_field",
+            "field_type": "VARCHAR(255)",
+            "required": True
+        }
+        logger.info(f"Testing invalid customer GUID with data: {data}")
+        response = requests.post(url, json=data)
+        self.assertEqual(response.status_code, 400, "Invalid customer_guid should result in 400")
+        self.assertIn("Database", response.text)
+        logger.info("Test case for invalid customer_guid passed.")
+
+    def test_add_duplicate_custom_field_name(self):
+        """Test adding a custom field with a duplicate field name."""
+        url = f"{self.BASE_URL}/custom_fields"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "duplicate_field",
+            "field_type": "VARCHAR(255)",
+            "required": True
+        }
+
+        # Add the custom field for the first time
+        logger.info(f"Adding initial custom field with data: {data}")
+        response = requests.post(url, json=data)
+        self.assertEqual(response.status_code, 200, "Failed to add the initial custom field")
+
+        # Try adding the same custom field again
+        logger.info(f"Testing duplicate custom field name with data: {data}")
+        response = requests.post(url, json=data)
```

Comment:
Lets give out 200, if field_name, field_type and required field matches for duplicate IDs
If any of the above fields does not match then we can say a field_name already exists with appropriate HTTP error

Reply:
modified sir